server/remote/msgraph/remote.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package msgraph
+
+import (
+	"context"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/microsoft"
+
+	msgraph "github.com/yaegashi/msgraph.go/v1.0"
+
+	"github.com/mattermost/mattermost-plugin-msoffice/server/config"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/remote"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/utils"
+)
+
+const Kind = "msgraph"
+
+type impl struct {
+	conf   *config.Config
+	logger utils.Logger
+}
+
+func init() {
+	remote.Makers[Kind] = NewRemote
+}
+
+func NewRemote(conf *config.Config, logger utils.Logger) remote.Remote {
+	return &impl{
+		conf:   conf,
+		logger: logger,
+	}
+}
+
+// NewMicrosoftGraphClient creates a new client.
+func (r *impl) NewClient(ctx context.Context, token *oauth2.Token) remote.Client {
```

Comment:
```suggestion
func (r *impl) NewMicrosoftGraphClient(ctx context.Context, token *oauth2.Token) remote.Client {
```

or change comment to say `NewClient`